service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/dataset/DatasetCloneInputs.java

```diff
@@ -0,0 +1,50 @@
+package bio.terra.workspace.service.resource.controlled.flight.clone.dataset;
+
+import java.util.UUID;
+
+public class DatasetCloneInputs {
+  private final UUID workspaceId;
+  private final String projectId;
+  private final String datasetName;
+
+  public DatasetCloneInputs(UUID workspaceId, String projectId, String datasetName) {
+    this.workspaceId = workspaceId;
+    this.projectId = projectId;
+    this.datasetName = datasetName;
+  }
+
+  public UUID getWorkspaceId() {
+    return workspaceId;
+  }
+
+  public String getProjectId() {
+    return projectId;
+  }
+
+  public String getDatasetName() {
+    return datasetName;
+  }
+
+  @Override
+  public String toString() {
+    return "DatasetCloneInputs{"
+        + "workspaceId="
+        + workspaceId
+        + ", projectId='"
+        + projectId
+        + '\''
+        + ", datasetName='"
+        + datasetName
+        + '\''
+        + '}';
+  }
+
+  /**
+   * The dataset ID includes the project name.
+   *
+   * @return
+   */
+  public String getDatasetId() {
```

Comment:
If you serialize this class, you will get a datasetId field. For derived fields, it would be good hygiene to add a `@JsonIgnore` to the getter.